### PR TITLE
:rocket: feat:  `ERC4626` integration 

### PR DESCRIPTION
**Pull request by halaprix**
## Ticket URL

Please provide a link to the ticket:

## Description of Changes

Please list the changes introduced by this PR:

- add `ERC4626` support under `common`
- add `ERC4626Deposit`, `ERC4626Withdraw`, `MorphoBlueClaimRewards`, `RetunrMultipleTokens` action contracts
- added operations for deposit and withdrawal for `ERC4626`
- added strategies for deposit and withdrawal for `ERC4626`
- added operation to claim and return multiple `MorphoBlue` rewards - using multiple calls to Universal Reward Distributors
- added a generic position `view` for `ERC4626` vaults
- added e2e tests ( skipped due to one inch configuration issues - can be run locally )

## How to Test

Please provide instructions on how to test the changes in this PR:

## PR Definition of Done

Please ensure the following requirements have been met before marking the PR as ready for review:

- [ ] All checks are passing
- [ ] PR is linked to a corresponding ticket
- [ ] PR title is clear and concise
- [ ] Code has been self-reviewed and any fixes or improvements noted (See Code review standards in Notion)
- [ ] Documentation has been updated if necessary

### CODE REVIEW

File: packages/dma-contracts/contracts/actions/common/ERC4626Withdraw.sol [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    IERC4626 vault = IERC4626(depositData.vault);
+    uint beforeWithdrawal = IERC20(vault.asset()).balanceOf(address(this));
+
+    if (depositData.amount == type(uint256).max) {
+      uint256 maxWithdraw = vault.maxWithdraw(address(this));
+      vault.withdraw(maxWithdraw, address(this), address(this));
+    } else {
+      vault.withdraw(mappedWithdrawAmount, address(this), address(this));
+    }
+    uint256 withrawnAmount = IERC20(vault.asset()).balanceOf(address(this)) - beforeWithdrawal;
```
Comment (BackToTheCode) [inline packages/dma-contracts/contracts/actions/common/ERC4626Withdraw.sol:44]: You want this value to be negative?

File: packages/dma-contracts/hardhat.config.base.ts [hunk 2]
```diff
@@ -167,10 +167,10 @@ const config = {
         },
       ],
       chainId: ChainIdByNetwork[Network.LOCAL],
-      mining: {
-        auto: true,
-        interval: 2000,
-      },
+      // mining: {
```
Comment (jakubswierczek) [inline packages/dma-contracts/hardhat.config.base.ts]: can we uncomment it?

File: packages/dma-contracts/package.json [hunk 3]
```diff
@@ -20,7 +20,7 @@
     "lint": "yarn eslint . --ext .ts --max-warnings=0 && prettier --list-different 'contracts/**/*.sol'",
     "test": "yarn test:unit && yarn test:integration:nocompile && yarn test:e2e:nocompile",
     "test:nocompile": "yarn test:unit:nocompile && yarn test:integration:nocompile && yarn test:e2e:nocompile",
-    "test:e2e": "yarn clean && yarn compile && npx hardhat test --grep E2E",
+    "test:e2e": "npx hardhat test --grep E2E",
```
Comment (jakubswierczek) [inline packages/dma-contracts/package.json]: There is `test:e2e:nocompile` right under that.

Comment (halaprix) [reply packages/dma-contracts/package.json]: nonsense :joy:

Comment (halaprix) [reply packages/dma-contracts/package.json]: reverting -pushed by mistake :)

**Review by jakubswierczek — APPROVED** [review]

**Pull request merged**